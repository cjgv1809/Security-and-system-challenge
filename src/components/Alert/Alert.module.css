.successMessage {
  color: var(--color-success);
  font-size: 1rem;
}

.alert {
  padding: 0.625rem 0.9375rem;
  border-radius: 4px;
  margin-bottom: 0.9375rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.message {
  margin: 0;
}

.closeButton {
  background: none;
  border: none;
  font-size: 1.5rem;
  cursor: pointer;
  padding: 0;
  line-height: 1;
}

.success {
  background-color: var(--color-alert-success-background);
  color: var(--color-alert-success-text);
  border: 1px solid var(--color-alert-success-border);
}

.error {
  background-color: var(--color-alert-error-background);
  color: var(--color-alert-error-text);
  border: 1px solid var(--color-alert-error-border);
}

.info {
  background-color: var(--color-alert-info-background);
  color: var(--color-alert-info-text);
  border: 1px solid var(--color-alert-info-border);
}

.warning {
  background-color: var(--color-alert-warning-background);
  color: var(--color-alert-warning-text);
  border: 1px solid var(--color-alert-warning-border);
}
