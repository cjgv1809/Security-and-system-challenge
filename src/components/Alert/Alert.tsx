"use client";

import React, { useEffect } from "react";
import styles from "./Alert.module.css";
import type { AlertType } from "@/app/types";

type AlertProps = {
  type: AlertType;
  message: string;
  onClose?: () => void;
};

const Alert: React.FC<AlertProps> = ({ type, message, onClose }) => {
  useEffect(() => {
    if (onClose) {
      const timer = setTimeout(() => {
        onClose();
      }, 5000);

      return () => {
        clearTimeout(timer);
      };
    }
  }, [onClose]);

  return (
    <div className={`${styles.alert} ${styles[type]}`} role="alert">
      <p className={styles.message}>{message}</p>
      {onClose && (
        <button
          type="button"
          className={styles.closeButton}
          onClick={onClose}
          aria-label="Cerrar alerta"
        >
          &times;
        </button>
      )}
    </div>
  );
};

export default Alert;
