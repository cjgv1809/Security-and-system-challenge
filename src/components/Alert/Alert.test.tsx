import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "./Alert";

describe("Alert", () => {
  it("renders the alert with the message", () => {
    const message = "This is an alert";
    render(<Alert type="info" message={message} />);

    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it("calls onClose after 5 seconds", () => {
    jest.useFakeTimers();
    const onClose = jest.fn();
    render(<Alert type="info" message="This is an alert" onClose={onClose} />);

    jest.advanceTimersByTime(5000);

    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Alert type="info" message="This is an alert" />
    );

    expect(container).toMatchSnapshot();
  });
});
