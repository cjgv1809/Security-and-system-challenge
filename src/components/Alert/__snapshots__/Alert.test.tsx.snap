// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<div>
  <div
    class="alert info"
    role="alert"
  >
    <p
      class="message"
    >
      This is an alert
    </p>
  </div>
</div>
`;
